### PR TITLE
OP-1167 Add error for unexpected condition on lot management

### PR DESCRIPTION
**Pull request by mwithi**
See OP-1167.

Paired with https://github.com/informatici/openhospital-gui/pull/1935.

### CODE REVIEW

File: src/main/java/org/isf/medicalstock/service/MedicalStockIoOperations.java [hunk 1]
```diff
@@ -96,12 +100,17 @@ public List<Integer> getMedicalsFromLot(String lotCode) throws OHServiceExceptio
...
 		int qty = movement.getQuantity(); // movement initial quantity
-		List<Movement> dischargingMovements = new ArrayList<>();
+
+		if (lots.isEmpty()) {
+			LOGGER.warn("No lots with available quantity found for medical {}", medical.getDescription());
+			return dischargingMovements;
+		}
 		for (Lot lot : lots) {
-			Movement splitMovement = new Movement(movement.getMedical(), movement.getType(), movement.getWard(),
+			Movement splitMovement = new Movement(medical, movement.getType(), movement.getWard(),
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/medicalstock/service/MedicalStockIoOperations.java:113]: `medical` is extracted into a local variable so it doesn't need to be fetched over and over; then why aren't `type` and `ward` done the same way?

**Review by dbmalkovsky — APPROVED** [review]
LGTM; given my comment is ignored.

File: src/main/java/org/isf/medicalstock/service/MedicalStockIoOperations.java (hunk 1 above)
Comment (mwithi) [reply src/main/java/org/isf/medicalstock/service/MedicalStockIoOperations.java:113]: `type` and `ward` are extracted only once. Maybe we can do it for `lot.getCode()` inside the `for` loop?

Comment (dbmalkovsky) [reply src/main/java/org/isf/medicalstock/service/MedicalStockIoOperations.java:113]: It is inside of a loop so for every iteration they are fetched, aren't they?

Comment (mwithi) [reply src/main/java/org/isf/medicalstock/service/MedicalStockIoOperations.java:113]: Yes, I mean extract the lot.getCode() at the begin of the loop

Comment (mwithi) [reply src/main/java/org/isf/medicalstock/service/MedicalStockIoOperations.java:114]: done anyway

**Pull request merged**